src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -150,50 +177,73 @@ public string GetServerUrl(CommandSettings command)
-            agent = await _agentServer.UpdateAgentAsync(poolId, agent);
-            await GetAndAddTags(agent, command);
+            var deploymentMachine = new DeploymentMachine() { Agent = agent };
+            deploymentMachine = await _deploymentGroupServer.ReplaceDeploymentMachineAsync(new Guid(agentSettings.ProjectId), agentSettings.DeploymentGroupId, agent.Id, deploymentMachine);
 
-            return agent;
+            await GetAndAddTags(deploymentMachine.Id, agentSettings, agent, command);
+            return deploymentMachine.Agent;
         }
 
-        public async Task<TaskAgent> AddAgentAsync(int poolId, TaskAgent agent, CommandSettings command)
+        public async Task<TaskAgent> AddAgentAsync(AgentSettings agentSettings, TaskAgent agent, CommandSettings command)
         {
-            agent = await _agentServer.AddAgentAsync(poolId, agent);
-            await GetAndAddTags(agent, command);
+            var deploymentMachine = new DeploymentMachine(){ Agent = agent };
+            deploymentMachine = await _deploymentGroupServer.AddDeploymentMachineAsync(new Guid(agentSettings.ProjectId), agentSettings.DeploymentGroupId, deploymentMachine);
+            
+            await GetAndAddTags(deploymentMachine.Id, agentSettings, deploymentMachine.Agent, command);
 
-            return agent;
+            return deploymentMachine.Agent;
         }
 
-        public Task DeleteAgentAsync(int agentPoolId, int agentId)
+        public async Task DeleteAgentAsync(AgentSettings agentSettings)
         {
-            return _agentServer.DeleteAgentAsync(agentPoolId, agentId);
+            string currentAction = StringUtil.Loc("UninstallingService");
+            var machines = await GetDeploymentMachinesAsync(agentSettings);
+            Trace.Verbose("Returns {0} machines with name {1}", machines.Count, agentSettings.AgentName);
+            var machine = machines.FirstOrDefault();
+            if (machine == null)
+            {
+                _term.WriteLine(StringUtil.Loc("Skipping") + currentAction);
+            }
+            else
+            {
+                if (!string.IsNullOrWhiteSpace(agentSettings.ProjectId))
+                {
+                    await _deploymentGroupServer.DeleteDeploymentMachineAsync(new Guid(agentSettings.ProjectId), agentSettings.DeploymentGroupId, machine.Id);
+                }
+                else
+                {
+                    await _deploymentGroupServer.DeleteDeploymentMachineAsync(agentSettings.ProjectName, agentSettings.DeploymentGroupId, machine.Id);
+                }
+                _term.WriteLine(StringUtil.Loc("Success") + currentAction);
+            }
         }
 
         public async Task TestConnectionAsync(string url, VssCredentials creds)
         {
             _term.WriteLine(StringUtil.Loc("ConnectingToServer"));
             VssConnection connection = ApiUtil.CreateConnection(new Uri(url), creds);
 
-            await _agentServer.ConnectAsync(connection);
```

Comment:
deploymentGroup do not need connection at account level. It will have its own connection till project level